extensions/positron-assistant/src/copilot.ts

```diff
@@ -0,0 +1,334 @@
+}
+
+/** Register the Copilot service. */
+export function registerCopilotService(context: ExtensionContext) {
+	const copilotService = CopilotService.create(context);
+	context.subscriptions.push(copilotService);
+}
+
+export class CopilotService implements vscode.Disposable {
+	private readonly _disposables: vscode.Disposable[] = [];
+
+	/** The CopilotService singleton instance. */
+	private static _instance?: CopilotService;
+
+	private _client?: CopilotLanguageClient;
+
+	/** Create the CopilotLanguageService singleton instance. */
+	public static create(context: ExtensionContext) {
+		if (CopilotService._instance) {
+			throw new Error('CopilotService was already created.');
+		}
+		CopilotService._instance = new CopilotService(context);
+		return CopilotService._instance;
+	}
+
+	/** Retrieve the CopilotLanguageService singleton instance. */
+	public static instance(): CopilotService {
+		if (!CopilotService._instance) {
+			throw new Error('CopilotService was not created. Call create() first.');
+		}
+		return CopilotService._instance;
+	}
+
+	private constructor(
+		private readonly _context: vscode.ExtensionContext,
+	) {
+		this.registerCommands();
+	}
+
+	/** Register Copilot commands. */
+	private registerCommands() {
+		this._disposables.push(
+			vscode.commands.registerCommand('positron-assistant.copilot.signin', async () => {
+				await this.signIn();
+			}),
+			vscode.commands.registerCommand('positron-assistant.copilot.signout', async () => {
+				await this.signOut();
+			})
+		);
+	}
+
+	/** Get the Copilot language client. */
+	private client(): CopilotLanguageClient {
+		if (!this._client) {
+			// The client does not exist, create it.
+			const serverName = platform() === 'win32' ? 'copilot-language-server.exe' : 'copilot-language-server';
+			const command = path.join(this._context.extensionPath, 'resources', 'copilot', serverName);
+			const executable: Executable = {
+				command,
+				args: ['--stdio'],
+				transport: TransportKind.stdio,
+			};
+			const packageJSON = this._context.extension.packageJSON;
+			const editorPluginInfo: EditorPluginInfo = {
+				name: packageJSON.name,
+				version: packageJSON.version,
+			};
+			this._client = new CopilotLanguageClient(executable, editorPluginInfo);
+		}
+		return this._client;
+	}
+
+	/**
+	 * Prompt the user to sign in to Copilot if they aren't already signed in.
+	 */
+	private async signIn(): Promise<void> {
+		// HACK: Register the Copilot completion item provider.
+		// This is a temporary workaround until the configuration UI supports
+		// Copilot completions. It should be safe for now since the sign in
+		// command is only enabled when the hidden `positron.assistant.copilot.enable`
+		// setting is enabled.
+		this.registerInlineCompletionItemProvider();
+
+		const client = this.client();
+		const response = await client.sendRequest(SignInRequest.type, {});
+
+		if ('status' in response && 'user' in response) {
+			return;
+		}
```

Comment:
Could we show a notification toast or something similar to confirm that we're already signed in and completions are enabled? My workaround was to check the output pane to see if completions were occurring.